src/Microsoft.ML.TorchSharp/AutoFormerV2/ObjectDetectionTrainer.cs

```diff
@@ -0,0 +1,945 @@
+                ScoreColumnName = ctx.LoadString(),
+                NumberOfClasses = ctx.Reader.ReadInt32(),
+            };
+
+            var ch = env.Start("Load Model");
+
+            var model = new AutoFormerV2(options.NumberOfClasses,
+                embedChannels: new List<int>() { 64, 128, 256, 448 },
+                depths: new List<int>() { 2, 2, 6, 2 },
+                numHeads: new List<int>() { 2, 4, 8, 14 },
+                device: TorchUtils.InitializeDevice(env));
+
+            BinarySaver saver = new BinarySaver(env, new BinarySaver.Arguments());
+            DataViewType type;
+            object value;
+            env.CheckDecode(saver.TryLoadTypeAndValue(ctx.Reader.BaseStream, out type, out value));
+            var vecType = type as VectorDataViewType;
+            env.CheckDecode(vecType != null);
+            env.CheckDecode(value != null);
+            var labelGetter = Microsoft.ML.Internal.Utilities.Utils.MarshalInvoke(_decodeInitMethodInfo, vecType.ItemType.RawType, value);
+
+            var meta = new DataViewSchema.Annotations.Builder();
+            meta.Add(AnnotationUtils.Kinds.KeyValues, type, labelGetter);
+
+            var labelCol = new DataViewSchema.DetachedColumn(options.LabelColumnName, type, meta.ToAnnotations());
+
+            if (!ctx.TryLoadBinaryStream("TSModel", r => model.load(r)))
+                throw env.ExceptDecode();
+
+            return new ObjectDetectionTransformer(env, options, model, labelCol);
+        }
+
+        private static Delegate DecodeInit<T>(object value)
+        {
+            VBuffer<T> buffValue = (VBuffer<T>)value;
+            ValueGetter<VBuffer<T>> buffGetter = (ref VBuffer<T> dst) => buffValue.CopyTo(ref dst);
+            return buffGetter;
+        }
+
+        private protected class ObjDetMapper : MapperBase
+        {
+            private protected readonly ObjectDetectionTransformer Parent;
+            private protected readonly HashSet<int> InputColIndices;
+
+            private static readonly FuncInstanceMethodInfo1<ObjDetMapper, DataViewSchema.DetachedColumn, Delegate> _makeLabelAnnotationGetter
+                = FuncInstanceMethodInfo1<ObjDetMapper, DataViewSchema.DetachedColumn, Delegate>.Create(target => target.GetLabelAnnotations<int>);
+
+
+            public ObjDetMapper(ObjectDetectionTransformer parent, DataViewSchema inputSchema) :
+                base(Contracts.CheckRef(parent, nameof(parent)).Host.Register(nameof(ObjDetMapper)), inputSchema, parent)
+            {
+                Parent = parent;
+                InputColIndices = new HashSet<int>();
+                if (inputSchema.TryGetColumnIndex(parent.Options.BoundingBoxColumnName, out var col))
+                    InputColIndices.Add(col);
+
+                if (inputSchema.TryGetColumnIndex(parent.Options.ImageColumnName, out col))
+                    InputColIndices.Add(col);
+
+                torch.random.manual_seed(1);
```

Comment:
Should we respect seed from MLContext here

Reply:
I didn't before because we didn't have access to it, but with the changes I made now we can access it. So I have updated it to relfect the changes.